### PR TITLE
settings populated with backend

### PR DESCRIPTION
**Pull request by hanmosh**
- populate account settings page from MongoDB 
- can edit settings page (including tags) and it will save in the backend
- when you sign in or create account it will save your email in localStorage under "userEmail"

### CODE REVIEW

**Comment by ZealSV** [comment]
Nice work! Settings now responsive and communicative with backend, good endpoint creation.

**Comment by sanjanachecker** [comment]
Good job Hannah, no more hard coded info!

**Comment by hanmosh** [comment]
> Nice work! Settings now responsive and communicative with backend, good endpoint creation.

Thank you!

**Comment by hanmosh** [comment]
> Good job Hannah, no more hard coded info!

Yay! thanks

File: nextjs/src/app/settings/page.tsx [hunk 1]
```diff
@@ -9,18 +9,26 @@ import { useKindeAuth } from '@kinde-oss/kinde-auth-react';
 import majorsData from './majors.json';
 
 export default function Settings() {
-    const { logout } = useKindeAuth();
+    const { logout, user: authUser } = useKindeAuth();
+
+    const userId = '66627e51203c5440e5b9b46c'; // Replace this with the actual user ID
```
Comment (ObjectJosh) [inline nextjs/src/app/settings/page.tsx:14]: Make sure to replace this on prod!

File: nextjs/src/app/settings/page.tsx [hunk 2]
```diff
@@ -39,12 +47,142 @@ export default function Settings() {
...
+                if (!email) {
+                    console.error('No email found in localStorage');
+                    return;
+                }
+
+                const response = await fetch(`/api/users/${email}`);
+                const data = await response.json();
+                if (data.success) {
+                    const user = data.data;
+                    // setUserId(user._id); // Set the user ID
```
Comment (ObjectJosh) [inline nextjs/src/app/settings/page.tsx:72]: And to add this back

File: nextjs/src/app/settings/page.tsx [hunk 3]
```diff
@@ -39,12 +47,142 @@ export default function Settings() {
...
+    //         if (data.success) {
+    //             console.log('User data updated successfully:', data.data);
+    //         } else {
+    //             console.error('Failed to update user data:', data.error);
+    //         }
+    //     } catch (error) {
+    //         console.error('Error updating user data:', error);
+    //     }
+    // };
+
```
Comment (ObjectJosh) [inline nextjs/src/app/settings/page.tsx:185]: Clean up extra comments in the code

File: nextjs/src/app/tags/page.tsx [hunk 4]
```diff
@@ -44,28 +44,144 @@ const tags = {
 };
 
 const EditTags: React.FC = () => {
-    const [selectedTags, setSelectedTags] = useState<string[]>([]);
+    const { user: authUser } = useKindeAuth();
+    const userId = '66627e51203c5440e5b9b46c'; // Replace this with the actual user ID
```
Comment (ObjectJosh) [inline nextjs/src/app/tags/page.tsx:48]: Make sure to replace this on prod

File: nextjs/src/app/tags/page.tsx [hunk 5]
```diff
@@ -44,28 +44,144 @@ const tags = {
...
+    //             console.log('Tags updated successfully:', data.data);
+    //             router.push('/settings');
+    //         } else {
+    //             console.error('Failed to update tags:', data.error);
+    //         }
+    //     } catch (error) {
+    //         console.error('Error updating tags:', error);
+    //     }
+    // };
+
```
Comment (ObjectJosh) [inline nextjs/src/app/tags/page.tsx:178]: Same as above

**Review by ObjectJosh — COMMENTED** [review]
Looks good! Added a few comments about hardcoded user id & commented code 🫡

**Pull request merged**